### PR TITLE
Fix formatter reordering comments in function type parameters

### PR DESCRIPTION
**Pull request by robertDurst**
Address #5225 .

Specifically, this PR:

1. ensures that the positions of comments within nested function params are maintained by popping comments per argument rather than collecting all and appending at the end.
2. adds a unit test to reproduce bug and demonstrate it is fixed (with some bonus edge cases).
3. updates expected comment order in the test `comment_in_tuple_return_type` per conversations in #5225.

Tried my best to follow of convention. Long term goal here is to hopefully continue finding places to contribute, so nit picks of all sizes appreciated 😁

### CODE REVIEW

File: compiler-core/src/format/tests/function.rs [hunk 1]
```diff
@@ -343,12 +343,42 @@ fn comment_in_tuple_return_type() {
 }
 "#,
         r#"pub fn main() -> #(
-  String,
   // This is a string
+  String,
```
Comment (robertDurst) [inline compiler-core/src/format/tests/function.rs:347]: Test fix as per comment in conversation here: https://github.com/gleam-lang/gleam/issues/5225#issuecomment-3694874313

**Comment by ankddev** [comment]
Hey @robertDurst! Could you add a changelog entry (see existing entries for inspiration)?

**Comment by robertDurst** [comment]
> Hey @robertDurst! Could you add a changelog entry (see existing entries for inspiration)?

Done! Thanks for catching this @ankddev ... had the `CONTRIBUTING.md` open in a tab ... longingly waiting for me to read it 🤦

**Comment by robertDurst** [comment]
Ok, cleaned up the commit history, should be GTG for an initial review.

Thanks in advance and happy holidays 🥳 🎄

File: compiler-core/src/format.rs [hunk 2]
```diff
@@ -3049,6 +3065,30 @@ impl<'comments> Formatter<'comments> {
         }
     }
 
+    fn wrap_arguments_with_line_breaks<'a, I>(
```
Comment (lpil) [inline compiler-core/src/format.rs]: How come this is needed when none of the other matching syntaxes need this? Can we not use the same approach for all of them?

**Review by lpil — COMMENTED** [review]
Thank you!! I've left a small comment inline

File: compiler-core/src/format.rs (hunk 2 above)
Comment (robertDurst) [reply compiler-core/src/format.rs]: Thanks for the review! Let me take another pass here - had a nice few days off here on holiday, refreshed and with a clear mind 🧘

Comment (lpil) [reply compiler-core/src/format.rs]: Thank you 💜  Happy new year!

File: compiler-core/src/format.rs [hunk 3]
```diff
@@ -790,11 +790,38 @@ impl<'comments> Formatter<'comments> {
...
+                .append(
+                    printed_comments(self.pop_comments(location.end), false)
+                        .map_or(nil(), |c| line().append(c)),
+                )
+                .nest(INDENT)
+                .append(line())
+                .append(")")
+        } else {
+            self.wrap_arguments(arguments, location.end)
+        }
```
Comment (lpil) [inline compiler-core/src/format.rs:824]: It looks like this duplicates the code within `wrap_arguments`. Why is it that that code didn't do the job?

I think we should work out what the bug was there so it works for all `(...)`, rather than having a special case where the code is replicated here.

**Comment by robertDurst** [comment]
Optimistically started this during vacation and haven't had enough contiguous blocks of time since to develop a solid enough mental model to be effective. Closing in case someone else is interested in taking this on.
